src/execution_tree/compiler/compiler.cpp

```diff
@@ -359,7 +352,8 @@ namespace phylanx { namespace execution_tree { namespace compiler
                     std::move(name), external_variable(f, default_locality_));
 
                 static std::string define_variable("define-variable");
-                std::size_t sequence_number =
+                name_parts.primitive = define_variable;
```

Comment:
Same here, the static variable does not have any performance effect as the allocation is needed in any case.